### ISSUE TITLE
Kubernetes 1.22

### ISSUE THREAD

**Issue by bkalem** [cri-o/cri-o#5180]
Hello,

When i try to run kubernetes 1.22 with CRI-O 1.21 some error. 

Do you have any beta CRI-O 1.22 ?

Best regards

**Comment by haircommander** [comment]
I hope to have 1.22 cut some time next week. Until then, you can try installing cri-o from the main branch with https://github.com/cri-o/cri-o/blob/master/scripts/get

**Comment by haircommander** [comment]
mostly general fixes, though we should be adding swap support to conform to the latest cri changes

**Comment by AbdelmalekBen98** [comment]
i have the same problem , when i deployed nginx image in k8s 1.22 with cri-o 1.21 the status of containers was blocked in "Pending" , i didn't have this problem with k8s 1.21 and crio 1.21 . i supposed that the problem was in cri-o  . 
any suggestions pls ?
@MarcoR83  i have some questions if you don't mind .
have you changed the kubelet config ? 
and if it is possible can you share with me your kubelet config please ? 

Thank YOU ♥ .

**Comment by haircommander** [comment]
as an update: cri-o ci has been blocked the past week preventing us from merging some important pieces for the release. It is now fixed, and we will be ushering everything through asap :)

**Comment by MarcoR83** [comment]
> i have the same problem , when i deployed nginx image in k8s 1.22 with cri-o 1.21 the status of containers was blocked in "Pending" , i didn't have this problem with k8s 1.21 and crio 1.21 . i supposed that the problem was in cri-o .
> any suggestions pls ?
> @MarcoR83 i have some questions if you don't mind .
> have you changed the kubelet config ?
> and if it is possible can you share with me your kubelet config please ?
> 
> Thank YOU ♥ .


/etc/kubernetes/kubelet.env
```
KUBELET_ADDRESS="--node-ip=10.1.0.1"
KUBELET_HOSTNAME="--hostname-override=hostname1.domain.tld"

KUBELET_ARGS="--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf \
--config=/etc/kubernetes/kubelet-config.yaml \
--kubeconfig=/etc/kubernetes/kubelet.conf \
--container-runtime=remote \
--container-runtime-endpoint=unix:///var/run/crio/crio.sock \
--runtime-cgroups=/systemd/system.slice \
   --feature-gates=IPv6DualStack=true,SeccompDefault=true --seccomp-default "
KUBELET_NETWORK_PLUGIN="--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
```

what i added was --feature-gates=IPv6DualStack=true,SeccompDefault=true --seccomp-default 

/etc/kubernetes/kublet-config.yml
```
  apiVersion: kubelet.config.k8s.io/v1beta1
  kind: KubeletConfiguration
  nodeStatusUpdateFrequency: "10s"
  failSwapOn: True
  authentication:
    anonymous:
      enabled: false
    webhook:
      enabled: True
    x509:
      clientCAFile: /etc/kubernetes/ssl/ca.crt
  authorization:
    mode: Webhook
  staticPodPath: /etc/kubernetes/manifests
  cgroupDriver: systemd
  containerLogMaxFiles: 5
  containerLogMaxSize: 10Mi
  maxPods: 110
  address: 0.0.0.0
  readOnlyPort: 0
  healthzPort: 10248
  healthzBindAddress: 127.0.0.1
  kubeletCgroups: /systemd/system.slice
  clusterDomain: cluster.local
  protectKernelDefaults: true
  rotateCertificates: true
  clusterDNS:
  - 10.233.0.3
  kubeReserved:
    cpu: 200m
    memory: 512Mi
  resolvConf: "/run/systemd/resolve/resolv.conf"
  eventRecordQPS: 5
  shutdownGracePeriod: 60s
  shutdownGracePeriodCriticalPods: 20s
```

in kubespray I had to make sure that resolvConf: "/run/systemd/resolve/resolv.conf" is set, because dns was not working on Fedora 34 without. Might be the default for F34 not , not sure.

```
[crio]
log_dir = "/var/log/crio/pods"
version_file = "/var/run/crio/version"
version_file_persist = "/var/lib/crio/version"
[crio.api]
listen = "/var/run/crio/crio.sock"
stream_address = "127.0.0.1"
stream_port = "10010"
stream_enable_tls = false
stream_tls_cert = ""
stream_tls_key = ""
stream_tls_ca = ""
grpc_max_send_msg_size = 16777216
grpc_max_recv_msg_size = 16777216
[crio.runtime]
default_runtime = "crun"
no_pivot = false
decryption_keys_path = "/etc/crio/keys/"
conmon = "/usr/libexec/crio/conmon"
conmon_cgroup = "system.slice"
conmon_env = [
        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
]
default_env = [
]
selinux = true
seccomp_profile = "/etc/containers/seccomp.json"
seccomp_use_default_when_empty = true
cgroup_manager = "systemd"
default_capabilities = [
        "CHOWN",
        "DAC_OVERRIDE",
        "FSETID",
        "FOWNER",
        "SETGID",
        "SETUID",
        "SETPCAP",
        "NET_BIND_SERVICE",
        "KILL",
]
default_sysctls = [
        "net.ipv4.ping_group_range=0 2147483647",
]
additional_devices = [
]
hooks_dir = [
        "/usr/share/containers/oci/hooks.d",
]
default_mounts = [
]
pids_limit = 4096
log_size_max = -1
log_to_journald = false
container_exits_dir = "/var/run/crio/exits"
container_attach_socket_dir = "/var/run/crio"
bind_mount_prefix = ""
read_only = false
log_level = "info"
log_filter = ""
uid_mappings = ""
gid_mappings = ""
ctr_stop_timeout = 30
manage_ns_lifecycle = false
namespaces_dir = "/var/run"
pinns_path = ""
[crio.runtime.runtimes.runc]
runtime_path = "/usr/bin/runc"
runtime_type = "oci"
runtime_root = "/run/runc"
privileged_without_host_devices = false
[crio.runtime.runtimes.crun]
runtime_path = "/usr/bin/crun"
runtime_type = "oci"
runtime_root = "/run/crun"
privileged_without_host_devices = false
[crio.image]
default_transport = "docker://"
global_auth_file = ""
pause_image = "k8s.gcr.io/pause:3.3"
pause_image_auth_file = ""
pause_command = "/pause"
signature_policy = ""
insecure_registries = [
  ]
image_volumes = "mkdir"
registries = [
  ]
[crio.network]
network_dir = "/etc/cni/net.d/"
plugin_dirs = [
        "/opt/cni/bin",
        "/usr/libexec/cni",
]
[crio.metrics]
enable_metrics = false
metrics_port = 9090
```

this config I fiddled with in the past when building a 1.18 cluster and took it into a new fresh kubernetes 1.21 install which I then upgraded to 1.22 (using same cri.o 1.21.2) in an attempt to make it work with cgroupv2 (unified-thingy=1, which is default since F33 I think)

I did not use kubespray to deploy the container runtime. I installed it using this repos  build:
https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.21/Fedora_34/

that because cri-o team stopped doing "modular" builds in the Fedora Build System, which I,  after trying to deal with by myself , can relate to... because it is  a mess. Only problem is that I found out about this fact far later about the fact that they moved to OBS for their builds.  In the Fedora SRPM project sources I just read "orphaned" at one point and was wondering if anyone would take them over, and thought .. well I guess I have to maintain it myself.. until I found that OBS thing.
So  after I had figured that out I adapted and added the repo instead on relying on the (outdated, not  maintained) fedora builds.

Long story short, I made it work with this. Yet im looking forward to a matching 1.22 version of cri-o for my cluster :)

**Comment by AbdelmalekBen98** [comment]
> > i have the same problem , when i deployed nginx image in k8s 1.22 with cri-o 1.21 the status of containers was blocked in "Pending" , i didn't have this problem with k8s 1.21 and crio 1.21 . i supposed that the problem was in cri-o .
> > any suggestions pls ?
> > @MarcoR83 i have some questions if you don't mind .
> > have you changed the kubelet config ?
> > and if it is possible can you share with me your kubelet config please ?
> > Thank YOU ♥ .
> 
> /etc/kubernetes/kubelet.env
> 
> ```
> KUBELET_ADDRESS="--node-ip=10.1.0.1"
> KUBELET_HOSTNAME="--hostname-override=hostname1.domain.tld"
> 
> KUBELET_ARGS="--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf \
> --config=/etc/kubernetes/kubelet-config.yaml \
> --kubeconfig=/etc/kubernetes/kubelet.conf \
> --container-runtime=remote \
> --container-runtime-endpoint=unix:///var/run/crio/crio.sock \
> --runtime-cgroups=/systemd/system.slice \
>    --feature-gates=IPv6DualStack=true,SeccompDefault=true --seccomp-default "
> KUBELET_NETWORK_PLUGIN="--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
> ```
> 
> what i added was --feature-gates=IPv6DualStack=true,SeccompDefault=true --seccomp-default
> 
> /etc/kubernetes/kublet-config.yml
> 
> ```
>   apiVersion: kubelet.config.k8s.io/v1beta1
>   kind: KubeletConfiguration
>   nodeStatusUpdateFrequency: "10s"
>   failSwapOn: True
>   authentication:
>     anonymous:
>       enabled: false
>     webhook:
>       enabled: True
>     x509:
>       clientCAFile: /etc/kubernetes/ssl/ca.crt
>   authorization:
>     mode: Webhook
>   staticPodPath: /etc/kubernetes/manifests
>   cgroupDriver: systemd
>   containerLogMaxFiles: 5
>   containerLogMaxSize: 10Mi
>   maxPods: 110
>   address: 0.0.0.0
>   readOnlyPort: 0
>   healthzPort: 10248
>   healthzBindAddress: 127.0.0.1
>   kubeletCgroups: /systemd/system.slice
>   clusterDomain: cluster.local
>   protectKernelDefaults: true
>   rotateCertificates: true
>   clusterDNS:
>   - 10.233.0.3
>   kubeReserved:
>     cpu: 200m
>     memory: 512Mi
>   resolvConf: "/run/systemd/resolve/resolv.conf"
>   eventRecordQPS: 5
>   shutdownGracePeriod: 60s
>   shutdownGracePeriodCriticalPods: 20s
> ```
> 
> in kubespray I had to make sure that resolvConf: "/run/systemd/resolve/resolv.conf" is set, because dns was not working on Fedora 34 without. Might be the default for F34 not , not sure.
> 
> /etc/crio/crio.conf
> 
> ```
> [crio]
> log_dir = "/var/log/crio/pods"
> version_file = "/var/run/crio/version"
> version_file_persist = "/var/lib/crio/version"
> [crio.api]
> listen = "/var/run/crio/crio.sock"
> stream_address = "127.0.0.1"
> stream_port = "10010"
> stream_enable_tls = false
> stream_tls_cert = ""
> stream_tls_key = ""
> stream_tls_ca = ""
> grpc_max_send_msg_size = 16777216
> grpc_max_recv_msg_size = 16777216
> [crio.runtime]
> default_runtime = "crun"
> no_pivot = false
> decryption_keys_path = "/etc/crio/keys/"
> conmon = "/usr/libexec/crio/conmon"
> conmon_cgroup = "system.slice"
> conmon_env = [
>         "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
> ]
> default_env = [
> ]
> selinux = true
> seccomp_profile = "/etc/containers/seccomp.json"
> seccomp_use_default_when_empty = true
> cgroup_manager = "systemd"
> default_capabilities = [
>         "CHOWN",
>         "DAC_OVERRIDE",
>         "FSETID",
>         "FOWNER",
>         "SETGID",
>         "SETUID",
>         "SETPCAP",
>         "NET_BIND_SERVICE",
>         "KILL",
> ]
> default_sysctls = [
>         "net.ipv4.ping_group_range=0 2147483647",
> ]
> additional_devices = [
> ]
> hooks_dir = [
>         "/usr/share/containers/oci/hooks.d",
> ]
> default_mounts = [
> ]
> pids_limit = 4096
> log_size_max = -1
> log_to_journald = false
> container_exits_dir = "/var/run/crio/exits"
> container_attach_socket_dir = "/var/run/crio"
> bind_mount_prefix = ""
> read_only = false
> log_level = "info"
> log_filter = ""
> uid_mappings = ""
> gid_mappings = ""
> ctr_stop_timeout = 30
> manage_ns_lifecycle = false
> namespaces_dir = "/var/run"
> pinns_path = ""
> [crio.runtime.runtimes.runc]
> runtime_path = "/usr/bin/runc"
> runtime_type = "oci"
> runtime_root = "/run/runc"
> privileged_without_host_devices = false
> [crio.runtime.runtimes.crun]
> runtime_path = "/usr/bin/crun"
> runtime_type = "oci"
> runtime_root = "/run/crun"
> privileged_without_host_devices = false
> [crio.image]
> default_transport = "docker://"
> global_auth_file = ""
> pause_image = "k8s.gcr.io/pause:3.3"
> pause_image_auth_file = ""
> pause_command = "/pause"
> signature_policy = ""
> insecure_registries = [
>   ]
> image_volumes = "mkdir"
> registries = [
>   ]
> [crio.network]
> network_dir = "/etc/cni/net.d/"
> plugin_dirs = [
>         "/opt/cni/bin",
>         "/usr/libexec/cni",
> ]
> [crio.metrics]
> enable_metrics = false
> metrics_port = 9090
> ```
> 
> this config I fiddled with in the past when building a 1.18 cluster and took it into a new fresh kubernetes 1.21 install which I then upgraded to 1.22 (using same cri.o 1.21.2) in an attempt to make it work with cgroupv2 (unified-thingy=1, which is default since F33 I think)
> 
> I did not use kubespray to deploy the container runtime. I installed it using this repos build:
> https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.21/Fedora_34/
> 
> that because cri-o team stopped doing "modular" builds in the Fedora Build System, which I, after trying to deal with by myself , can relate to... because it is a mess. Only problem is that I found out about the fact far later, that they moved to OBS for their builds. In the Fedora SRPM project sources I just read "orphaned" at one point and was wondering if anyone would take them over, and thought .. well I guess I have to maintain it myself.. until I found that OBS thing.
> So after I had figured that out I adapted and added the repo instead on relying on the (outdated, not maintained) fedora builds.
> 
> Long story short, I made it work with this. Yet im looking forward to a matching 1.22 version of cri-o for my cluster :)

Thank you @MarcoR83  ,i really appreciate your help ♥ ♥

**Comment by MarcoR83** [comment]
you welcome :)

**Comment by haircommander** [comment]
> In the Fedora SRPM project sources I just read "orphaned" at one point and was wondering if anyone would take them over, and thought .. well I guess I have to maintain it myself.. until I found that OBS thing.

FYI, I'm in the process of adopting the package :) . it was orphaned by mistake

**Comment by bkalem** [comment]
hello @haircommander 

Thank you for your get script, it's work nice : 

```shell
sudo dnf install -y curl jq tar intel-cmt-cat
sudo wget -O /tmp/crio-get-script.sh https://raw.githubusercontent.com/cri-o/cri-o/master/scripts/get
sudo bash /tmp/crio-get-script.sh

sudo systemctl daemon-reload
sudo systemctl enable --now crio 
```

```shell
[root@master-01 ~]# crio --version
WARN[0000] Encountered error detecting link mode of binary:
WARN[0000] Encountered error detecting link mode of binary:
INFO[0000] RDT is not enabled: failed to detect resctrl mount point: resctrl not found in /proc/mounts
crio version 1.22.0
Version:          1.22.0
GitCommit:        71ab70084e36b4c46bec1bdab7c6fbc840585740
GitTreeState:     dirty
BuildDate:        1980-01-01T00:00:00Z
GoVersion:        go1.16.6
Compiler:         gc
Platform:         linux/amd64
Linkmode:         unknown
BuildTags:        apparmor, exclude_graphdriver_devicemapper, seccomp, selinux
SeccompEnabled:   true
AppArmorEnabled:  false
```

**Comment by bkalem** [comment]
hello,

i report the error `error detecting link mode of binary` in this link : 
https://github.com/cri-o/cri-o/issues/5165#issuecomment-901131450

**Comment by MarcoR83** [comment]
Any news on the release?

**Comment by MarcoR83** [comment]
Question: whats with cri-o 1.23. No package available for Fedora 35 yet? 
![image](https://user-images.githubusercontent.com/53475938/158586247-43fc51ab-a13a-4874-9275-735d8ebc491b.png)
I read a while ago somewhere (dont remember where )that there was an issue because of a new Go version used for 1.23. Is this still an issue? If so why cant a new version of go be used to build the module for F35? 

It seams cri-o packages are always troublesome and behind unfortunately :(

**Comment by MarcoR83** [comment]
Neither Fedora itself nor  OBS have rpms for F35.. 
![image](https://user-images.githubusercontent.com/53475938/158589729-e77ed081-0ea1-429d-b142-4c6e55ee3b71.png)

**Comment by haircommander** [comment]
hey @MarcoR83 sorry, F35 doesn't support go 1.17 which cri-o 1.23 requires. I can enable in OBS

**Comment by haircommander** [comment]
hm looks like https://build.opensuse.org/project/show/devel:kubic:libcontainers:stable:cri-o:1.23 should have F35 enabled, would that work for you?
